CONTRIBUTING.rst

```diff
@@ -3,45 +3,98 @@
 How to contribute to nilearn
 =============================
 
-This project is a community effort, and everyone is welcome to
-contribute.
-
-The project is hosted on https://github.com/nilearn/nilearn
+This project, hosted on https://github.com/nilearn/nilearn, is a community
+effort, and everyone is welcome to contribute. We value very much your feedback
+and opinion on features that should be improved or added. All discussions
+are public and hold on relevant issues or pull requests. To discuss your matter,
+please answer a relevant `issue <https://github.com/nilearn/nilearn/issues>`_
+or open a new one.
 
 The best way to contribute and to help the project is to start working on known
-issues.
-See `Good first issues <https://github.com/nilearn/nilearn/labels/Good%20first%20issue>`_ to get
-started.
+issues such as `Good first issues <https://github.com/nilearn/nilearn/labels/Good%20first%20issue>`_ ,
+`known bugs <https://github.com/nilearn/nilearn/labels/Bug>`_ or
+`proposed enhancements <https://github.com/nilearn/nilearn/labels/Enhancement>`_.
 
 If an issue does not already exist for a potential contribution, we ask that
 you first open an `issue <https://github.com/nilearn/nilearn/issues>`_ before
-sending a :ref:`pull request`.
+sending a :ref:`pull request` to discuss in advance scope and potential design
+choices.
+
+
+Contribution Guidelines
+---------------------------
+
+We ask every contributor to respect common guidelines. Those are inspired from
+`scikit-learn
+<https://scikit-learn.org/stable/developers/contributing.html#contributing-code>`_
+and ensure Nilearn remains simple to understand, efficient and maintanable.
+For example tests need to run quickly in order not to burden the development process.
+To keep continuous integration efficient with our limited infrastructure, running
+all the examples must lead to downloading a limited amount of data (gigabytes)
+and execute in a reasonable amount of time (a few hours).
+
+Those guidelines will hence be checked during reviewing process.
+
+
++--------------+-------------+---------------------------------------+
+|              | Which PR ?  |        Guidelines                     |
++==============+=============+=======================================+
+|              |             | - Clearly showcase benefits           |
+|  Examples    | New features| - Run in a few minutes                |
+|              |             | - Use light data from Nilearn         |
+|              |             | - Renders well after build            |
++--------------+-------------+---------------------------------------+
+|              |             | - Test type is adapted to behavior    |
+|              |             | - Tests pass continuous integration   |
+|              |  Bugfixes   | - Doesn't decrease coverage           |
```

Comment:
Bugfixes seems to be rendered oddly.
![image](https://user-images.githubusercontent.com/8228902/114210021-4a0ee400-992d-11eb-957c-9e0fb8e3ee2b.png)